src/JustEat.StatsD/TagsFormatters/StatsDTagsFormatter.cs

```diff
@@ -0,0 +1,172 @@
+            _tagsSeparator = configuration.TagsSeparator?.ToArray() ?? Array.Empty<char>();
+            _keyValueSeparator = configuration.KeyValueSeparator?.ToArray() ?? Array.Empty<char>();
+            _prefixSize = Encoding.UTF8.GetByteCount(_prefix);
+            _suffixSize = Encoding.UTF8.GetByteCount(_suffix);
+            _tagsSeparatorSize = Encoding.UTF8.GetByteCount(_tagsSeparator);
+            _keyValueSeparatorSize = Encoding.UTF8.GetByteCount(_keyValueSeparator);
+        }
+        
+        /// <inheritdoc />
+        public bool AreTrailing { get; }
+
+        /// <inheritdoc />
+        public virtual int GetTagsBufferSize(in Dictionary<string, string?> tags)
+        {
+            const int NoTagsSize = 0;
+            if (!AreTagsPresent(tags))
+            {
+                return NoTagsSize;
+            }
+
+            return _prefixSize
+                + GetTagsSize(tags)
+                + _suffixSize;
+        }
+        
+        /// <inheritdoc />
+        public virtual ReadOnlySpan<char> FormatTags(in Dictionary<string, string?> tags)
+        {
+            if (!AreTagsPresent(tags))
+            {
+                return ReadOnlySpan<char>.Empty;
+            }
+
+            char[] destination = Buffer;
+            if (!TryFormatTags(destination, tags, out int written))
+            {
+                var newSize = GetTagsBufferSize(tags);
+                _buffer = new char[newSize];
+                destination = Buffer;
+
+                if (!TryFormatTags(destination, tags, out written))
+                {
+                    throw new FormatException("Failed to format tags.");
+                }
+            }
+
+            return new ArraySegment<char>(destination, 0, written);
+        }
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool TryFormatTags(Span<char> destination, in Dictionary<string, string?> tags, out int written)
+        {
+            var buffer = new Buffer<char>(destination);
+            bool isFormattingSuccessful = buffer.TryWrite(_prefix)
+                && TryWriteTags(ref buffer, tags)
+                && buffer.TryWrite(_suffix);
+
+            written = isFormattingSuccessful ? buffer.Written : 0;
+
+            return isFormattingSuccessful;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool TryWriteTags(ref Buffer<char> src, in Dictionary<string, string?> tags)
+        {
+            using var enumerator = tags.GetEnumerator();
+            var index = 0;
+            bool isFormattingSuccessful = true;
+            while (isFormattingSuccessful && enumerator.MoveNext())
+            {
+                isFormattingSuccessful = TryWriteTag(ref src, enumerator.Current)
+                    && TryWriteTagsSeparator(ref src, index, tags);
+
+                index++;
+            }
+
+            return isFormattingSuccessful;
+        }
```

Comment:
```suggestion
        private bool TryWriteTags(ref Buffer<char> src, in Dictionary<string, string?> tags)
        {
            var index = 0;
            foreach (var tag in tags)
            {
                var isFormattingSuccessful = TryWriteTag(ref src, tag)
                                             && TryWriteTagsSeparator(ref src, index++, tags);
                if (!isFormattingSuccessful)
                {
                    return false;
                }
            }
            return true;
        }
```

Reply:
Done! I have renamed the `Buffer` variable from `src` to `buffer`.